x-pack/plugins/ml/server/lib/ml_client/ml_client.ts

```diff
@@ -0,0 +1,399 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ElasticsearchClient, IScopedClusterClient } from 'kibana/server';
+
+import { JobSavedObjectService, JobType } from '../../saved_objects';
+
+import {
+  Job,
+  JobStats,
+  Datafeed,
+  DatafeedStats,
+} from '../../../common/types/anomaly_detection_jobs';
+import { Calendar } from '../../../common/types/calendars';
+import { searchProvider } from './search';
+
+import { DataFrameAnalyticsConfig } from '../../../common/types/data_frame_analytics';
+import { MLJobNotFound } from './errors';
+
+type OrigMlClient = ElasticsearchClient['ml'];
+
+export interface MlClient extends OrigMlClient {
+  anomalySearch: ReturnType<typeof searchProvider>['anomalySearch'];
+}
+
+export function getMlClient(
+  client: IScopedClusterClient,
+  jobSavedObjectService: JobSavedObjectService
+): MlClient {
+  const mlClient = client.asInternalUser.ml;
+
+  async function jobIdsCheck(jobType: JobType, p: any) {
+    const jobIds = getADJobIds(p);
+    if (jobIds.length) {
+      const filteredJobIds = await jobSavedObjectService.filterJobIdsForSpace(jobType, jobIds);
+      const missingIds = jobIds.filter((j) => filteredJobIds.indexOf(j) === -1);
+      if (missingIds.length) {
+        throw new MLJobNotFound(`No known job with id '${missingIds.join(',')}'`);
+      }
+    }
+  }
+
+  async function datafeedIdsCheck(p: any) {
+    const datafeedIds = getDatafeedIds(p);
+    if (datafeedIds.length) {
+      const filteredDatafeedIds = await jobSavedObjectService.filterDatafeedIdsForSpace(
+        datafeedIds
+      );
+      const missingIds = datafeedIds.filter((j) => filteredDatafeedIds.indexOf(j) === -1);
+      if (missingIds.length) {
+        throw new MLJobNotFound(`No known job with id '${missingIds.join(',')}'`);
```

Comment:
Just checking, is this text correct in that it says `job` rather than `datafeed`?